pkg/operation/botanist/secrets.go

```diff
@@ -988,40 +1024,79 @@ func (b *Botanist) generateBasicAuthAPIServer(ctx context.Context, existingSecre
 func (b *Botanist) generateStaticToken(ctx context.Context, existingSecretsMap map[string]*corev1.Secret) (*secrets.StaticToken, error) {
 	staticTokenConfig := &secrets.StaticTokenSecretConfig{
 		Name: common.StaticTokenSecretName,
-		Tokens: []secrets.TokenConfig{
-			{
+		Tokens: map[string]secrets.TokenConfig{
+			common.KubecfgUsername: {
 				Username: common.KubecfgUsername,
 				UserID:   common.KubecfgUsername,
 				Groups:   []string{user.SystemPrivilegedGroup},
 			},
-			{
+			common.KubeAPIServerHealthCheck: {
 				Username: common.KubeAPIServerHealthCheck,
 				UserID:   common.KubeAPIServerHealthCheck,
 			},
-		},
-	}
+			common.KonnectivityServerUserName: {
+				Username: common.KonnectivityServerUserName,
+				UserID:   common.KonnectivityServerUserName,
+			},
+		}}
 
+	var (
+		newStaticTokenConfig = secrets.StaticTokenSecretConfig{
+			Name:   common.StaticTokenSecretName,
+			Tokens: make(map[string]secrets.TokenConfig),
+		}
+		staticToken *secrets.StaticToken
+	)
 	if existingSecret, ok := existingSecretsMap[staticTokenConfig.Name]; ok {
-		staticToken, err := secrets.LoadStaticTokenFromCSV(staticTokenConfig.Name, existingSecret.Data[secrets.DataKeyStaticTokenCSV])
+		var err error
+		staticToken, err = secrets.LoadStaticTokenFromCSV(staticTokenConfig.Name, existingSecret.Data[secrets.DataKeyStaticTokenCSV])
 		if err != nil {
 			return nil, err
 		}
 
-		b.mutex.Lock()
-		defer b.mutex.Unlock()
+		var tokenConfigSet, tokenSet = sets.NewString(), sets.NewString()
+		for _, t := range staticTokenConfig.Tokens {
+			tokenConfigSet.Insert(t.Username)
+		}
+		for _, t := range staticToken.Tokens {
+			tokenSet.Insert(t.Username)
+		}
 
-		b.Secrets[staticTokenConfig.Name] = existingSecret
+		if diff := tokenConfigSet.Difference(tokenSet); diff.Len() > 0 {
+			for {
+				if diff.Len() == 0 {
+					break
+				}
 
-		if err := b.storeAPIServerHealthCheckToken(staticToken); err != nil {
-			return nil, err
-		}
+				tokenKey, _ := diff.PopAny()
+				newStaticTokenConfig.Tokens[tokenKey] = staticTokenConfig.Tokens[tokenKey]
```

Comment:
😕 What is happening here? Do you want to loop over the diff? Why not a regular `for ... := range diff.UnsortedList() {}`?

Reply:
because I forgot we have this luxurious call :D